vendor/github.com/fatih/color/color.go

```diff
@@ -108,7 +116,14 @@ const (
 
 // New returns a newly created color object.
 func New(value ...Attribute) *Color {
-	c := &Color{params: make([]Attribute, 0)}
+	c := &Color{
+		params: make([]Attribute, 0),
+	}
+
+	if noColorExists() {
+		c.noColor = boolPtr(true)
+	}
```

Comment:
Honestly, the fact that they touch `os` and the fact that they don't really give you a choice whether you want to respect `NO_COLOR` or not is more than a bit shitty... I get that it makes the library API slightly easier to use, and it "just works" for the majority of users, but it makes testing things that use this library (like k6) such a PITA... :sob: 

At this point, I'd happily remove it and replace it with something else, since we moved the end-of-test summary to JS, we use it very sparingly:
```
rg -g '!vendor' -g '!go.mod' -g '!go.sum' 'fatih/color|color\.'
ui/form.go
27:	"github.com/fatih/color"
50:		if _, err := fmt.Fprintln(w, color.BlueString(f.Banner)+"\n"); err != nil {
60:				displayLabel += " " + color.New(color.Faint, color.FgCyan).Sprint("["+extra+"]")
66:			color.Set(color.FgCyan)
73:			color.Unset()
80:				if _, printErr := fmt.Fprintln(w, color.RedString("- "+err.Error())); printErr != nil {

ui/pb/progressbar.go
28:	"github.com/fatih/color"
34:	colorFaint   = color.New(color.Faint)
35:	statusColors = map[Status]*color.Color{
36:		Interrupted: color.New(color.FgRed),
37:		Done:        color.New(color.FgGreen),

cmd/cloud.go
37:	"github.com/fatih/color"
333:			valueColor := getColor(noColor || !stdoutTTY, color.FgCyan)

cmd/ui.go
35:	"github.com/fatih/color"
89:func getColor(noColor bool, attributes ...color.Attribute) *color.Color {
91:		c := color.New()
96:	c := color.New(attributes...)
102:	c := getColor(noColor, color.FgCyan)
140:	valueColor := getColor(noColor, color.FgCyan)

cmd/login_cloud.go
29:	"github.com/fatih/color"
145:				valueColor := getColor(noColor || !stdoutTTY, color.FgCyan)
```

Not now, but the next time we do something with colors in k6...

Reply:
I did test and it definitely doesn't work ... hence #2091 was made ;) So this update has no consequences for us ... it appears. I am pretty sure it's because of https://github.com/k6io/k6/blob/04c8b1fcf9a7e8628c1200603d433b404df241ba/cmd/ui.go#L86-L99

Reply:
Glad to see my workaround continues to work around the issue :tada:  :sweat_smile: True, we won't support `NO_COLOR` until we explicitly add support for it, but I'd still prefer that approach :man_shrugging: